### PR TITLE
Add an `externalLink` field to Programs.

### LINKED ISSUE #607: Add "program details" URL for every program
On the program list page, we can offer a button that will let users open up a separate page to get more details about the program -- often, this will be a seattle.gov page. We would implement this by adding a field for a program details URL in each program.

This isn't required for launch by any means, but it would be a nice add if we have time to improve the UX.

### PR DESCRIPTION
**Pull request by AWBaumann**
### Description
Added a new field that lets admins specify an external link for additional program details.

### Checklist
- [X] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Fixes #607

### CODE REVIEW

File: browser-test/src/admin_program_translation.test.ts [hunk 1]
```diff
@@ -61,6 +61,10 @@ describe('Admin can manage translations', () => {
     const applicantQuestions = new ApplicantQuestions(page);
     await applicantQuestions.validateHeader('es-US');
 
+    
+    // Expect program details link to contain 'Detalles del programa' with link to 'http://seattle.gov'
```
Comment (carolinedanzi) [inline browser-test/src/admin_program_translation.test.ts:65]: praise: thanks for adding a test for this here, too!

File: browser-test/src/support/admin_programs.ts [hunk 2]
```diff
@@ -34,6 +34,10 @@ export class AdminPrograms {
     await this.page.fill('#program-display-name-input', programName);
     await this.page.fill('#program-display-description-textarea', description);
 
+    if (externalLink !== '') {
```
Comment (carolinedanzi) [inline browser-test/src/support/admin_programs.ts:37]: question: is this check necessary? We should be able to submit the empty string in the form without storing it, I think

File: universal-application-tool-0.0.1/app/controllers/dev/DatabaseSeedController.java [hunk 3]
```diff
@@ -192,7 +192,12 @@ private ProgramDefinition insertProgramWithBlocks(String name) {
     try {
       ProgramDefinition programDefinition =
           programService
-              .createProgramDefinition(name, "desc", name, "display description")
+              .createProgramDefinition(
+                  name,
+                  "desc",
+                  name,
+                  "display description",
+                  "https://github.com/seattle-uat/civiform")
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/controllers/dev/DatabaseSeedController.java:200]: praise: nice link choice :)

File: universal-application-tool-0.0.1/app/models/Program.java [hunk 4]
```diff
@@ -36,6 +36,9 @@
 
   @Constraints.Required private String description;
 
+  /** Link to external site for this program. */
+  @Constraints.Required private String externalLink;
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/models/Program.java:40]: question: should this be required? Or is it ok for a program to not have an external link?

File: universal-application-tool-0.0.1/app/models/Program.java [hunk 5]
```diff
@@ -36,6 +36,9 @@
 
   @Constraints.Required private String description;
 
+  /** Link to external site for this program. */
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/models/Program.java:39]: question (blocking): I'm so nervous for database updates - do we know what will happen to programs that do not currently have external links?

**Review by carolinedanzi — COMMENTED** [review]
This looks really nice! Just want some reassurance on the db update before approving

File: universal-application-tool-0.0.1/app/models/Program.java (hunk 5 above)
Comment (AWBaumann) [reply universal-application-tool-0.0.1/app/models/Program.java:39]: The evolutions update should cover this. Any programs that don't currently have external links will default to the empty string.

`alter table programs add column external_link varchar default '';`

Then in `ProgramIndexView` if the external link is empty then we fall back to the application info page.

File: universal-application-tool-0.0.1/app/models/Program.java (hunk 4 above)
Comment (AWBaumann) [reply universal-application-tool-0.0.1/app/models/Program.java:40]: I could go either way but I prefer required. Let me know if you think I should change it.

File: browser-test/src/support/admin_programs.ts (hunk 2 above)
Comment (AWBaumann) [reply browser-test/src/support/admin_programs.ts]: Done.

File: universal-application-tool-0.0.1/app/models/Program.java (hunk 5 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/models/Program.java:39]: Thanks Aaron!

File: universal-application-tool-0.0.1/app/models/Program.java (hunk 4 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/models/Program.java:40]: Sounds good to me!

**Review by carolinedanzi — APPROVED** [review]

**Pull request merged**